### PR TITLE
[skyrl-train] Add example for on-policy distillation

### PR DESCRIPTION
**Pull request by erictang000**
# Overview
Adds an example of extending the `RayPPOTrainer` to do on policy distillation by providing a custom `apply_reward_kl_penalty` function, a pass through advantage estimator, and using the `importance_sampling` loss function as detailed in the [thinky blog](https://tinker-docs.thinkingmachines.ai/losses#policy-gradient-importance_sampling).

### distilling rl trained qwen3-4b-base (dapo recipe) back into qwen3-4b-base
image

### CODE REVIEW

File: skyrl-train/retry.sh
Comment (SumanthRH) [inline skyrl-train/retry.sh:1]: Remove?

File: skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  trainer.eval_interval=5 \
+  trainer.update_epochs_per_batch=1 \
+  trainer.train_batch_size=1024 \
+  trainer.policy_mini_batch_size=256 \
+  trainer.micro_forward_batch_size_per_gpu=2 \
+  trainer.micro_train_batch_size_per_gpu=1 \
+  trainer.ckpt_interval=10 \
+  trainer.max_prompt_length=2048 \
+  generator.sampling_params.max_generate_length=8192 \
+  trainer.policy.optimizer_config.lr=1.0e-6 \
```
Comment (SumanthRH) [inline skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: Thinky loss function uses a plain`.sum()` and doesn't normalize by sequence length. We normalize by max sequence length. Should the LR change? https://tinker-docs.thinkingmachines.ai/losses#policy-gradient-importance_sampling .

File: skyrl-train/retry.sh
Comment (erictang000) [reply skyrl-train/retry.sh:1]: yes will do

File: skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh (hunk 1 above)
Comment (erictang000) [reply skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: hmm yeah you're right i guess it makes sense to not have to normalize by sequence length for per token rewards as well. So the code should be:

```python
torch.mean(torch.sum(loss * loss_mask, dim=-1))
```

Comment (erictang000) [reply skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: couldn't find details on learning rate in their post, I assume they kept it constant relative to the pure RL runs

Comment (erictang000) [reply skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: jk: 5e-5 was used: https://github.com/thinking-machines-lab/tinker-cookbook/tree/main/tinker_cookbook/recipes/distillation#on-policy-distillation

Comment (SumanthRH) [reply skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: Resolved offline. Seems like they implicitly normalize by max tokens somewhere because a plain sum doesn't make much sense with this LR

File: skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
+set -x
+
+# Running on policy distillation for Math on GSM8K
+# Uses Qwen-3-1.7B-Base as the student model and an RL trained Qwen-3-4B as the teacher model
```
Comment (tyler-griggs) [inline skyrl-train/examples/on_policy_distillation/run_on_policy_distill_math.sh]: I believe this example is not using 1.7B, right?
